rust/connlib/clients/shared/src/control.rs

```
@@ -280,7 +281,7 @@ impl<CB: Callbacks + 'static> ControlPlane<CB> {
                         .tunnel
                         .callbacks()
                         .on_error(&Error::ControlProtocolError);
```

FYI: This is an error that we want to surface to the user, that the given resource is not available at the moment and they should contact the system administrator.

---

hmm, this error in particular is that the reference doesn't exists it's a control protocol error, in the sense we're recieving something completely unexpected from the portal.

We can try to surface to the user that a particular resource is offline but we can only do that if we do have that resource id.

---

@conectado This is the error I was hitting today when accessing a resource that has no online gateways serving it. Maybe it should be a different error case?

---

Yeah we totally should not just silently fail if there is resource without online relays or gateways (the portal will send offline error).